torch/csrc/jit/script/compiler.cpp

```diff
@@ -1673,13 +1693,16 @@ std::vector<Value*> inlineCallTo(Graph& g, Graph& callee, ArrayRef<Value*> input
   return outputs;
 }
 
-void defineMethodsInModule(Module & m, const std::vector<Def>& definitions, const std::vector<Resolver>& resolvers, SugaredValuePtr self) {
+void defineMethodsInModule(Module & m, const std::vector<DefAndTypes>& definitions, const std::vector<Resolver>& resolvers, SugaredValuePtr self, bool pure_func) {
   FunctionTable table;
   JIT_ASSERT(definitions.size() == resolvers.size());
+  if (pure_func) {
+    JIT_ASSERT(definitions.size() == 1);
+  }
   auto resolver_it = resolvers.begin();
   std::vector<Method*> methods;
-  for(Def def : definitions) {
-    const std::string& name = def.name().name();
+  for(DefAndTypes def : definitions) {
+    const std::string& name = pure_func ? "forward" : def.def.name().name();
```

Comment:
Note: if DefAndTypes had a FunctionSchema, then you can use its name to decide the method name. That way you can parse `def foo(...)` but then register it as 'forward' by providing a schema with a different name. This will also fix the need for the `pure_func` flag.